library/core/src/slice/mod.rs

```diff
@@ -2599,6 +2599,32 @@ impl<T> [T] {
         }
     }
 
+    /// Fills `self` with elements returned by calling a closure repeatedly.
+    ///
+    /// This method uses a closure to create new values. If you'd rather
+    /// [`Clone`] a given value, use [`fill`]. If you want to use the [`Default`]
+    /// trait to generate values, you can pass [`Default::default`] as the
+    /// argument.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_fill_with)]
+    ///
+    /// let mut buf = vec![1; 10];
+    /// buf.fill_with(Default::default);
```

Comment:
I believe we should show an example rather than using `Default` here, since usually one could use `fill` if they could use `Default`. Probably we want an example that keeps changing?

Reply:
I think a simple example is fine. It might even be good to show that you can call this with `Default::default`, and don't need to do `|| Default::default()`.

Reply:
I adapted this example from `Vec::resize_with`. Given these methods are similar, using similar examples doesn't seem like a bad idea.